packages/react-button/src/components/Button/useButton.ts

```diff
@@ -1,36 +1,57 @@
 import * as React from 'react';
-import { makeMergeProps, resolveShorthandProps } from '@fluentui/react-utilities';
-import { useButtonState } from './useButtonState';
-import type { ButtonProps, ButtonShorthandPropsCompat, ButtonState } from './Button.types';
+import { resolveShorthand } from '@fluentui/react-utilities';
+import type { ButtonProps, ButtonSlots, ButtonState } from './Button.types';
+import { useARIAButton } from '@fluentui/react-aria';
 
 /**
  * Consts listing which props are shorthand props.
  */
-export const buttonShorthandPropsCompat: ButtonShorthandPropsCompat[] = ['icon'];
-
-const mergeProps = makeMergeProps<ButtonState>({ deepMerge: buttonShorthandPropsCompat });
+export const buttonSlots: (keyof ButtonSlots)[] = ['icon', 'root'];
 
 /**
  * Given user props, defines default props for the Button, calls useButtonState, and returns processed state.
  * @param props - User provided props to the Button component.
  * @param ref - User provided ref to be passed to the Button component.
  */
-export const useButton = (props: ButtonProps, ref: React.Ref<HTMLElement>, defaultProps?: ButtonProps): ButtonState => {
-  const state = mergeProps(
-    {
-      ref,
-      as: 'button',
-      // Slots
-      icon: { as: 'span' },
-      // Non-slot props
-      size: 'medium',
-      type: 'button', // This is added because the default for type is 'submit'
+export const useButton = (
+  {
+    icon,
+    block = false,
+    circular = false,
+    disabledFocusable = false,
+    iconPosition = 'before',
+    outline = false,
+    primary = false,
+    size = 'medium',
+    subtle = false,
+    transparent = false,
+    ...props
+  }: ButtonProps,
+  ref: React.Ref<unknown>,
+): ButtonState => {
+  const iconShorthand = resolveShorthand(icon);
+  return {
+    components: {
+      root: 'button',
+      icon: 'span',
     },
-    defaultProps && resolveShorthandProps(defaultProps, buttonShorthandPropsCompat),
-    resolveShorthandProps(props, buttonShorthandPropsCompat),
-  );
-
-  useButtonState(state);
-
-  return state;
+    root: useARIAButton(props, {
+      required: true,
+      defaultProps: {
+        ref: ref as React.Ref<HTMLButtonElement>,
+        type: 'button', // This is added because the default for type is 'submit'
+      },
+    }),
+    icon: iconShorthand,
+    block,
+    circular,
+    disabledFocusable,
+    iconPosition,
+    outline,
+    primary,
+    size,
+    subtle,
+    transparent,
+    iconOnly: Boolean(iconShorthand?.children && props.children),
```

Comment:
There is a separate manual accessibility review that we are going to do in all of our components. Since this PR is already quite big I'd like to include something like this separately if possible.